rp2040-hal/src/sio.rs

```diff
@@ -667,3 +676,224 @@ pub unsafe fn spinlock_reset() {
+    pub signed: bool,
+    #[doc = "Bits 10:14 - The most-significant bit allowed to pass by the mask (inclusive)  
+    Setting MSB < LSB may cause chip to turn inside-out"]
+    pub mask_msb: u8,
+    #[doc = "Bits 5:9 - The least-significant bit allowed to pass by the mask (inclusive)"]
+    pub mask_lsb: u8,
+    #[doc = "Bits 0:4 - Logical right-shift applied to accumulator before masking"]
+    pub shift: u8,
+}
+
+impl LaneCtrl {
+    /// Default value where all fields are zero and false
+    pub const DEFAULT: Self = Self {
+        blend: false,
+        force_msb: 0,
+        add_raw: false,
+        cross_result: false,
+        cross_input: false,
+        signed: false,
+        mask_msb: 0,
+        mask_lsb: 0,
+        shift: 0,
+    };
+
+    /// encode the configuration to be loaded in the ctrl register of one lane of an interpolator
+    pub const fn encode(&self) -> u32 {
+        assert!(self.force_msb < 0b100);
+        assert!(self.mask_msb < 0b100000);
+        assert!(self.mask_lsb < 0b100000);
+        assert!(self.mask_msb >= self.mask_lsb);
+        assert!(self.shift < 0b100000);
+        ((self.blend as u32) << 21)
+            | ((self.force_msb as u32) << 19)
+            | ((self.add_raw as u32) << 18)
+            | ((self.cross_result as u32) << 17)
+            | ((self.cross_input as u32) << 16)
+            | ((self.signed as u32) << 15)
+            | ((self.mask_msb as u32) << 10)
+            | ((self.mask_lsb as u32) << 5)
+            | (self.shift as u32)
+    }
+}
+
+///Trait representing the functionnality of a single lane of an interpolator.
+pub trait Lane {
+    ///Read the lane result, and simultaneously write lane results to both accumulators.
+    fn pop(&mut self) -> u32;
+    ///Read the lane result without altering any internal state
+    fn peek(&self) -> u32;
+    ///Write a value to the accumulator
+    fn set_accum(&mut self, v: u32);
+    ///Read the value from the accumulator
+    fn get_accum(&self) -> u32;
+    ///Write a value to the base register
+    fn set_base(&mut self, v: u32);
+    ///Read the value from the base register
+    fn get_base(&self) -> u32;
+    ///Write to the control register
+    fn set_ctrl(&mut self, v: u32);
+    ///Read from the control register
+    fn get_ctrl(&self) -> u32;
+    ///Add the value to the accumulator register
+    fn add_accum(&mut self, v: u32);
+    ///Read the raw shift and mask value (BASE register not added)
+    fn read_raw(&self) -> u32;
+}
+
+///Trait representing the functionnality of an interpolator.
+/// ```no_run
+/// use rp2040_hal::sio::{Sio,LaneCtrl,Lane};
+/// use rp2040_hal::pac;
+/// let mut peripherals = pac::Peripherals::take().unwrap();
+/// let mut sio = Sio::new(peripherals.SIO);
+///
+/// // by having the configuration const, the validity is checked during compilation.
+/// const config: u32 = LaneCtrl {
+///     mask_msb: 4, // Most significant bit is the fourt
```

Comment:
```suggestion
///     mask_msb: 4, // Most significant bit is the fourth
```
Nitpick: I wonder if 4th is the best way to describe a bit. While it feels ok to number bits 0,1,2,3,4..., personally I'd say something like "bit 0 is the first bit, bit 1 is the second bit....". So the fourth bit would be bit 3.

Reply:
I reworded the whole comment, I hope it is clearer like this